sympy/physics/quantum/qapply.py

```diff
@@ -68,18 +80,22 @@ def qapply(e, **options):
         |k><b|
         >>> qapply(A * b.dual / (b * b.dual))
         |k>
-        >>> qapply(k.dual * A / (k.dual * k), dagger=True)
-        <b|
         >>> qapply(k.dual * A / (k.dual * k))
-        <k|*|k><b|/<k|k>
+        <b|
     """
     from sympy.physics.quantum.density import Density
 
     dagger = options.get('dagger', False)
     sum_doit = options.get('sum_doit', False)
+    ip_doit = options.get('ip_doit', True)
 
-    if e == 0:
-        return S.Zero
+    if isinstance(e, (int, complex, float)):
+        e = sympify(e)
```

Comment:
Why not just call `sympify` unconditionally? Actually `_sympify` is better because it doesn't parse strings.

Reply:
Will fix this now.